### PR TITLE
Refactor DataVisualizer + test

### PR DESCRIPTION
**Pull request by ajpal**
# Description
- Only parse records when they change, not every time the component updates. I'm going to use this pattern for other data operations, too, for example determining which columns are numeric.
- Only call updateChart if the modal is open and there is a chart type selected (otherwise there's nothing to update)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/dataBrowser/DataVisualizer.jsx [hunk 1]
```diff
@@ -56,21 +57,71 @@ class DataVisualizer extends React.Component {
...
+          parsedRecords.push(JSON.parse(record));
+        }
+      });
+      return parsedRecords;
     }
   };
 
+  componentDidUpdate(previousProps, prevState) {
+    if (this.props !== previousProps) {
+      this.setState({parsedRecords: this.parseRecords()});
```
Comment (jmkulwik) [inline apps/src/storage/dataBrowser/DataVisualizer.jsx:120]: This should only happen when this.props.tableRecords changes, right? Can you make the if statement check for this specifically?

Comment (jmkulwik) [reply apps/src/storage/dataBrowser/DataVisualizer.jsx:120]: Also, how often do we expect this to happen? The component will keep updating even if it isn't visible. Is it worth checking if the component is visible and only running parseRecords if the vis is open?

Comment (ajpal) [reply apps/src/storage/dataBrowser/DataVisualizer.jsx:120]: Fixed. It ended being a kind of verbose conditional, so I used some variables to make it more clear

**Review by jmkulwik — APPROVED** [review]
Thanks for the fix!

**Pull request merged**